Training24/Test.c

```diff
@@ -1,14 +1,91 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A6 branch.
 // ------------------------------------------------------------------------------------------------
+_CRT_SECURE_NO_WARNINGS;
 #include <stdio.h>
-#include "Header.h"
+#include <corecrt.h>
+#include <io.h>
+#include <fcntl.h>
+#include <conio.h>
+#include <ctype.h>
+
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+// External declarations for functions
+extern void PrintChessboard (FILE* output);
+
+///<summary>Compares two files character by character and returns the result of the comparison.</summary>
+static int FileCompare (FILE* ref, FILE* output, int* row, int* col) {
+   wchar_t refChar = getwc (ref);
+   wchar_t outChar = getwc (output);
+   *row = 1;
+   *col = 0;
+   // Loop through both files character by character
+   while (refChar != WEOF && outChar != WEOF) {
+      *col += 1;
+      if (refChar != outChar) return -3;  // Mismatch found
+      // If newline is encountered, move to the next row
+      if (refChar == L'\n') {
+         *row += 1;
+         *col = 0;
+      }
+      // Get the next character from both files
+      refChar = getwc (ref);
+      outChar = getwc (output);
+   }
+   // To check if both files reached EOF at the same time
+   if (refChar != outChar) return refChar != WEOF ? -1 : -2;

```

Comment:
You should check this before hand using file length, not after parsing the whole file. It is more performant that way.